src/Processors/QtiV2/Marshallers/LearnosityObjectMarshaller.php

```diff
@@ -48,7 +48,9 @@ protected function marshallChildrenKnown(QtiComponent $component, array $element
                 $fragment->appendXML(QtiMarshallerUtil::marshallCollection(ContentCollectionBuilder::buildFlowCollectionContent($component->getComponents())));
                 $element = self::getDOMCradle()->createElement('div');
                 $element->setAttribute('data-type', 'sharedpassage');
-                $element->appendChild($fragment);
+                if($fragment->hasChildNodes()) {
```

Comment:
Formatting: If you could also fix the missing space between the `if` and `(`, that would be great.

Reply:
Done